sdks/typescript/src/v1/workflow.ts

```diff
@@ -0,0 +1,252 @@
+/* eslint-disable no-dupe-class-members */
+import WorkflowRunRef from '@hatchet/util/workflow-run-ref';
+import { Context } from '@hatchet/step';
+import { CronWorkflows, ScheduledWorkflows } from '@hatchet/clients/rest/generated/data-contracts';
+import { Workflow as WorkflowV0 } from '@hatchet/workflow';
+import { IHatchetClient } from './client/client.interface';
+import { CreateTaskOpts } from './task';
+
+const UNBOUND_ERR = new Error('workflow unbound to hatchet client, hint: use client.run instead');
+
+/**
+ * Additional metadata that can be attached to a workflow run.
+ */
+type AdditionalMetadata = Record<string, string>;
+
+/**
+ * Options for running a workflow.
+ */
+export type RunOpts = {
+  /**
+   * Additional metadata to attach to the workflow run.
+   */
+  additionalMetadata?: AdditionalMetadata;
+};
+
+/**
+ * Extracts a property from an object type based on task name, or falls back to inferred type
+ */
+type TaskOutputType<K, TaskName extends string, InferredType> = TaskName extends keyof K
+  ? K[TaskName]
+  : InferredType;
+
+/**
+ * Options for creating a new workflow.
+ */
+export type CreateWorkflowOpts = {
+  /**
+   * The name of the workflow.
+   */
+  name: WorkflowV0['id'];
+  /**
+   * Optional description of the workflow.
+   */
+  description?: WorkflowV0['description'];
+  /**
+   * Optional version of the workflow.
+   */
+  version?: WorkflowV0['version'];
+  /**
+   * Optional sticky strategy for the workflow.
+   */
+  sticky?: WorkflowV0['sticky'];
+  /**
+   * Optional schedule timeout for the workflow.
+   */
+  scheduleTimeout?: WorkflowV0['scheduleTimeout'];
+  /**
+   * Optional on config for the workflow.
+   */
+  on?: WorkflowV0['on'];
```

Comment:
another small one - I can change the `on_crons`, `on_events`, etc. to this - WDYT? seems like another easy way to have the APIs be closer together. that'd be good anyways since we could pull some more validation into Pydantic

Reply:
can we register both crons and events on the same workflow? part of this single "on" design gave us type saftey to only allow one or the other

Reply:
ah I'm not sure! I assumed we could

Reply:
lets test this but leave it for now